bioptim/optimization/optimal_control_program.py

```diff
@@ -380,7 +380,17 @@ def __init__(
         # Prepare the variable mappings
         if variable_mappings is None:
             variable_mappings = BiMappingList()
-        NLP.add(self, "variable_mappings", variable_mappings, True)
+
+        variable_mappings_per_phase = [{} for _ in range(self.n_phases)]
```

Comment:
That is not exactly right, but I understand why you did that... variable_mappings is not a dict, but an augmented dict (dict with extra features). Since it is hard to know if everything is declared or not, you felt that filling a parallel structure would make sense, and I mostly agree, but not in this case. 
Here we are going to simplify. Either ALL the mappings are declared with phase=-1, then we just duplicate them OR ALL the mappings are already declared for all the variables (meaning as soon as one phase!=-1 then all the variables must be declared by the user).

In other word, we only fill the easy case where the user only sent phase=-1 and don't do anything otherwise (apart maybe raising an error).

That said, we can be a bit more gentle and allow for specific variable to behave on their own (for instance "q" being declared with phase=-1, but "qdot" being declared phase=0, 1, ...). But doing the case where "q" is declared with phase=-1 and phase= 3, 4 and automatically complete the 0, 1, 2 and 5 for a 6 phases will be too complicated